cli/app/app.go

```diff
@@ -1259,44 +1276,26 @@ func Up(c *cli.Context) {
 		komposeObject = loadComposeFile(inputFile)
 	}
 
-	// Convert komposeObject to K8S controllers
-	mServices, mDeployments, _, _, _, _ := komposeConvert(komposeObject, opt)
-
-	// submit svc first
-	sc := &api.Service{}
-	for k, v := range mServices {
-		err := json.Unmarshal(v, &sc)
-		if err != nil {
-			logrus.Fatalf("Failed to unmarshal %s to service object: %s\n", k, err)
-		}
-		//submit sc to k8s
-		scCreated, err := client.Services(api.NamespaceDefault).Create(sc)
-		if err != nil {
-			logrus.Fatalf("Failed to create service %s: ", k, err)
-		} else {
-			fmt.Printf("Service %q has been created.\n", k)
-		}
-		logrus.Debugf("%s\n", scCreated)
-	}
+	//Convert komposeObject to K8S controllers
+	objects := komposeConvert(komposeObject, opt)
 
-	// then submit dc
-	dc := &extensions.Deployment{}
-	for k, v := range mDeployments {
-		err := json.Unmarshal(v, &dc)
-		if err != nil {
-			logrus.Fatalf("Failed to unmarshal %s to deployment controller object: %s\n", k, err)
-		}
-		//submit sc to k8s
-		dcCreated, err := client.Deployments(api.NamespaceDefault).Create(dc)
-		if err != nil {
-			logrus.Fatalf("Failed to create deployment controller %s: ", k, err)
-		} else {
-			fmt.Printf("Deployment %q has been created.\n", k)
+	for _, v := range objects {
+		switch t := v.(type) {
+		case *extensions.Deployment:
+			_, err := client.Deployments(api.NamespaceDefault).Create(t)
+			if err != nil {
+				logrus.Fatalf("Error creating deployment: %s", t.Name)
+			}
+			logrus.Infof("Successfully created deployment: %s", t.Name)
+		case *api.Service:
+			_, err := client.Services(api.NamespaceDefault).Create(t)
+			if err != nil {
+				logrus.Fatalf("Error creating service: %s", t.Name)
+			}
+			logrus.Infof("Successfully created service: %s", t.Name)
```

Comment:
Agreed we have to deploy svc first. Otherwise it is possible in some cases end up with application that is not working properly.